fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryBaseTest.java

```diff
@@ -48,20 +57,18 @@ public abstract class FDBDirectoryBaseTest {
 
     @BeforeEach
     public void setUp() {
-        if (fdb == null) {
-            fdb = FDBDatabaseFactory.instance().getDatabase();
-        }
-        if (subspace == null) {
-            subspace = fdb.run(context -> TestKeySpace.getKeyspacePath("record-test", "unit", "indexTest", "version").toSubspace(context));
-        }
-        fdb.run(context -> {
-            context.ensureActive().clear(subspace.range());
-            return null;
-        });
+        fdb = dbExtension.getDatabase();
+        path = pathManager.createPath(TestKeySpace.RAW_DATA);
         FDBRecordContext context = fdb.openContext(getContextConfig());
+        subspace = fdb.run(path::toSubspace);
         directory = new FDBDirectory(subspace, context, null);
     }
 
+    @AfterEach
+    public void tearDown() {
+        directory.getCallerContext().close();
```

Comment:
This was exposed by the new `FDBDatabaseExtension`'s check for unopened contexts (when I ran the test in a mode that tracked all open contexts)

Reply:
It creates the `context` in `setUp`, so changing that to a field instead of a local variable seems better than using `getCallerContext()` here.

Reply:
Sure